From e6295f25eabb9fcc8765561825ab7c9b124c1568 Mon Sep 17 00:00:00 2001
Message-ID: <e6295f25eabb9fcc8765561825ab7c9b124c1568.1707763691.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Mon, 5 Feb 2024 17:30:31 +0000
Subject: [PATCH v2 01/30] gnu: Add (gnu packages golang-build) module.

* gnu/packages/golang-build.scm: New file.
* gnu/local.mk: Add to gnu/packages/golang-build.scm scope.

Change-Id: I4605d44452859b47ed6369a234c7ead496b19c2a
---
 gnu/local.mk                  |  1 +
 gnu/packages/golang-build.scm | 44 +++++++++++++++++++++++++++++++++++
 2 files changed, 45 insertions(+)
 create mode 100644 gnu/packages/golang-build.scm

diff --git a/gnu/local.mk b/gnu/local.mk
index 7806771eea..e40be7fb46 100644
--- a/gnu/local.mk
+++ b/gnu/local.mk
@@ -307,6 +307,7 @@ GNU_SYSTEM_MODULES =				\
   %D%/packages/gnu-pw-mgr.scm			\
   %D%/packages/gobby.scm			\
   %D%/packages/golang.scm			\
+  %D%/packages/golang-build.scm		\
   %D%/packages/golang-check.scm		\
   %D%/packages/golang-crypto.scm	      	\
   %D%/packages/golang-web.scm	        	\
diff --git a/gnu/packages/golang-build.scm b/gnu/packages/golang-build.scm
new file mode 100644
index 0000000000..03fcdede80
--- /dev/null
+++ b/gnu/packages/golang-build.scm
@@ -0,0 +1,44 @@
+;;; GNU Guix --- Functional package management for GNU
+;;;
+;;; This file is part of GNU Guix.
+;;;
+;;; GNU Guix is free software; you can redistribute it and/or modify it
+;;; under the terms of the GNU General Public License as published by
+;;; the Free Software Foundation; either version 3 of the License, or (at
+;;; your option) any later version.
+;;;
+;;; GNU Guix is distributed in the hope that it will be useful, but
+;;; WITHOUT ANY WARRANTY; without even the implied warranty of
+;;; MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+;;; GNU General Public License for more details.
+;;;
+;;; You should have received a copy of the GNU General Public License
+;;; along with GNU Guix.  If not, see <http://www.gnu.org/licenses/>.
+
+(define-module (gnu packages golang-build)
+  #:use-module ((guix licenses) #:prefix license:)
+  #:use-module (guix build-system go)
+  #:use-module (guix gexp)
+  #:use-module (guix git-download)
+  #:use-module (guix packages)
+  #:use-module (guix utils)
+  #:use-module (gnu packages))
+
+;;; Commentary:
+;;;
+;;; Modules (libraries) which are part of the Golang project but outside the
+;;; main Golang tree, see <https://pkg.go.dev/golang.org/x>
+;;;
+;;; Since they are bound to be relied on by many, their dependencies should be
+;;; kept minimal, and this module should not depend on other modules
+;;; containing Golang packages.
+;;;
+;;; Please: Try to add new module packages in alphabetic order.
+;;;
+;;; Code:
+
+;;;
+;;; Avoid adding new packages to the end of this file. To reduce the chances
+;;; of a merge conflict, place them above by existing packages with similar
+;;; functionality or similar names.
+;;;

base-commit: bb92a4d7bc71197a51eea9075387c5313f299085
-- 
2.41.0

